src/datasets/iterable_dataset.py

```diff
@@ -9,7 +9,7 @@
 import numpy as np
 import pyarrow as pa
 
-from .arrow_dataset import DatasetInfoMixin
+from .arrow_dataset import DatasetInfoMixin, _align_features
```

Comment:
I think we can move `_align_features` (and `_check_if_features_can_be_aligned`) from `iterable_dataset.py`/`arrow_dataset.py` to `features.py` to have everything in one place